core/src/main/scala/cromwell/core/package.scala

```diff
@@ -1,17 +1,42 @@
 package cromwell
 
+import wdl4s._
+import wdl4s.expression.WdlFunctions
 import wdl4s.values.{SymbolHash, WdlValue}
 
+import scala.util.Try
 import scalaz._
 
 package object core {
   // root can be a Path instead of a String in PBE. stdout / err too but it doesn't really bring values since they're just stringified to WdlFiles
   class WorkflowContext(val root: String)
   class CallContext(override val root: String, val stdout: String, val stderr: String) extends WorkflowContext(root)
 
+  // Can evaluate a wdl value
+  class Evaluator(evaluator: WdlValue => Try[WdlValue]) {
+    def evaluate(wdlValue: WdlValue): Try[WdlValue] = evaluator(wdlValue)
```

Comment:
Yeah that could be interesting, I'm trying to make everything work in Cromwell first but when we have that we could definitely move some of this code to WDL4s